src/Cache/CollectionCacheKey.php

```diff
@@ -43,14 +43,14 @@ class CollectionCacheKey extends CacheKey
      * @param string               $association     The field name that represents the association.
      * @param array<string, mixed> $ownerIdentifier The identifier of the owning entity.
      */
-    public function __construct($entityClass, $association, array $ownerIdentifier)
+    public function __construct($entityClass, $association, array $ownerIdentifier, string $filterHash = '')
     {
         ksort($ownerIdentifier);
 
         $this->ownerIdentifier = $ownerIdentifier;
         $this->entityClass     = (string) $entityClass;
         $this->association     = (string) $association;
 
-        parent::__construct(str_replace('\\', '.', strtolower($entityClass)) . '_' . implode(' ', $ownerIdentifier) . '__' . $association);
+        parent::__construct(str_replace('\\', '.', strtolower($entityClass)) . '_' . implode(' ', $ownerIdentifier) . '__' . $association . '_' . $filterHash);
```

Comment:
I don't think there should be a trailing underscore if `$filterHash` is an empty string. It's going to have an impact on all existing entries for no good reason, and also, I think it will look ugly.